internal/resource-request-operator/broadcaster.go

```diff
@@ -0,0 +1,191 @@
+package resourcerequestoperator
+
+import (
+	"fmt"
+	"sync"
+
+	crdclient "github.com/liqotech/liqo/pkg/crdClient"
+
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+
+	configv1alpha1 "github.com/liqotech/liqo/apis/config/v1alpha1"
+	"github.com/liqotech/liqo/pkg/utils"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+)
+
+// Broadcaster is an object which is used to get resources of the cluster.
+type Broadcaster struct {
+	allocatable   corev1.ResourceList
+	clusterConfig configv1alpha1.ClusterConfig
+	offerMutex    sync.RWMutex
+	configMutex   sync.RWMutex
+}
+
+// SetupBroadcaster create the informer e run it to signal node changes updating Offers.
+func (b *Broadcaster) SetupBroadcaster(clientset *kubernetes.Clientset, config *rest.Config, stopper chan struct{}) error {
+	b.allocatable = corev1.ResourceList{}
+	factory := informers.NewSharedInformerFactory(clientset, 0)
+	informer := factory.Core().V1().Nodes().Informer()
+	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    b.onAdd,
+		UpdateFunc: b.onUpdate,
+		DeleteFunc: b.onDelete,
+	})
+
+	config.ContentConfig.GroupVersion = &configv1alpha1.GroupVersion
+	config.NegotiatedSerializer = scheme.Codecs.WithoutConversion()
+	config.UserAgent = rest.DefaultKubernetesUserAgent()
+	CRDclient, err := crdclient.NewFromConfig(config)
+	if err != nil {
+		return err
+	}
+	go utils.WatchConfiguration(b.setConfig, CRDclient, "")
```

Comment:
I would put this watch configuration outside the SetupBroadcaster, where other controllers do the same operation.